inception/inception-api-annotation-experimental/src/main/ts/client/actionhandling/AnnotationExperienceAPIActionHandler.ts

```diff
@@ -0,0 +1,109 @@
+
+    constructor(aAnnotationExperienceAPI: AnnotationExperienceAPIImpl) {
+        this.annotationExperienceAPI = aAnnotationExperienceAPI;
+    }
+
+
+    registerOnClickActionHandler(aTagName: string, aAction: string, aViewport: number[][]) {
+        let that = this;
+
+        let elem = document.querySelector("." + aTagName);
+        if (elem != null) {
+            //Click events on annotation page specific items
+            switch (aAction) {
+                case "select":
+                    for (let item of document.getElementsByClassName(aTagName)) {
+                        item.setAttribute("onclick", 'annotationExperienceAPI.sendSelectAnnotationMessageToServer(evt.target.attributes[3].value)');
+                    }
+                    break;
+                case "new_document":
+                    elem.addEventListener("click", () => {
+                        //that.annotationExperienceAPI.sendDocumentMessageToServer();
+                    });
+                    break;
+                case "viewport":
+                    elem.addEventListener("click", () => {
+                        //that.annotationExperienceAPI.sendViewportMessageToServer(aViewport);
+                    });
+                    break;
+                default:
+                    console.error("Can not register single click action, reason: Action-type not found.");
+                    return;
+            }
+
+            console.log("Action: " + aAction + " is registered for elements: " + aTagName)
+        } else {
+            console.error("Can not register single click action, reason: Element not found.");
+        }
+    }
+
+    registerOnDoubleClickActionHandler(aTagName: string, aAction: string) {
+        let elements = document.getElementsByClassName(aTagName);
+        if (elements != null) {
+            switch (aAction) {
+                case  "create":
+                    for (let item of elements) {
+                        item.setAttribute("ondblclick",
+                            'annotationExperienceAPI.sendCreateAnnotationMessageToServer(evt.target.attributes[3].value,' +
+                            'document.getElementsByClassName("dropdown")[0].children[1].getAttribute("title"),' +
+                            'evt.target.parentElement.attributes[1].value)');
+                    }
+                    break;
+                default:
+                    console.error("Can not register double click action, reason: Action-type not found.")
+                    return;
+            }
+        }
+
+        console.log("Action: " + aAction + " is registered for elements: " + aTagName)
+    }
+
+    registerDefaultActionHandler() {
+        //Click events on annotation page specific items
+        let that = this;
+        onclick = function (aEvent) {
+            let elem = <Element>aEvent.target;
+            console.log(elem)
+            if (elem.tagName === 'rect') {
```

Comment:
This code looks like it needs to know how rendering is done, so it should be part of an editor that is built *on* the API, but not as a part of the API itself - somebody could render not using SVG but using canvas or plain HTML - the API cannot make assumptions on that.